### PR TITLE
Be app 0173

### PR DESCRIPTION
**Pull request by habibdevsy**
- getOrderLogsByOrderId

### CODE REVIEW

File: backend/src/Manager/OrderLogs/OrderLogsManager.php [hunk 1]
```diff
@@ -31,4 +31,14 @@ public function createOrderLogs(OrderLogsCreateRequest $request): ?OrderLogsEnti
 
         return $entity;
     }
+
+    public function getOrderLogsByOrderId($orderId): ?array
```
Comment (ralsalloum) [inline backend/src/Manager/OrderLogs/OrderLogsManager.php]: type hint is missing for both new functions.

File: backend/src/Repository/OrderLogsEntityRepository.php [hunk 2]
```diff
@@ -19,32 +19,35 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, OrderLogsEntity::class);
     }
 
-    // /**
-    //  * @return OrderLogsEntity[] Returns an array of OrderLogsEntity objects
-    //  */
-    /*
-    public function findByExampleField($value)
+    public function getOrderLogsByOrderId($orderId): ?array
```
Comment (ralsalloum) [inline backend/src/Repository/OrderLogsEntityRepository.php]: type hint is missing.

File: backend/src/Repository/OrderLogsEntityRepository.php [hunk 3]
```diff
@@ -19,32 +19,35 @@ public function __construct(ManagerRegistry $registry)
...
             ->getQuery()
-            ->getResult()
-        ;
+            ->getResult();
     }
-    */
 
-    /*
-    public function findOneBySomeField($value): ?OrderLogsEntity
+    public function getCurrentStageDate($orderId): ?array
```
Comment (ralsalloum) [inline backend/src/Repository/OrderLogsEntityRepository.php]: type hint is missing.

File: backend/src/Response/OrderLogs/OrderLogTimeLineResponse.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Response\OrderLogs;
+
+class OrderLogTimeLineResponse
+{
+    public $completionTime;
```
Comment (ralsalloum) [inline backend/src/Response/OrderLogs/OrderLogTimeLineResponse.php]: type hint is missing for the first three vars.

File: backend/src/Service/DateFactory/DateFactoryService.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      if ($interval->y) { $result .= $interval->format("%y years "); }
+      if ($interval->m) { $result .= $interval->format("%m months "); }
+      if ($interval->d) { $result .= $interval->format("%d days "); }
+      if ($interval->h) { $result .= $interval->format("%h hours "); }
+      if ($interval->i) { $result .= $interval->format("%i minutes "); }
+      if ($interval->s) { $result .= $interval->format("%s seconds "); }
+
+      return $result;
+   }
+}
```
Comment (ralsalloum) [inline backend/src/Service/DateFactory/DateFactoryService.php:27]: New empty line at the end of the file is needed.

File: backend/src/Service/OrderLogs/OrderLogsService.php [hunk 6]
```diff
@@ -42,6 +48,52 @@ public function createOrderLogsRequest(OrderEntity $order, StoreOwnerBranchEntit
...
+
+            $orderStatus = $this->autoMapping->map('array', OrderLogTimeLineResponse::class, $state);
+
+            $response['orderState'] = $orderStatus ;
+            $response['logs'] = $orderLogs ;
+        }
+
+        return  $response;
+    }
+}
```
Comment (ralsalloum) [inline backend/src/Service/OrderLogs/OrderLogsService.php:99]: New empty line at the end of the file is needed.

File: backend/src/Repository/OrderLogsEntityRepository.php [hunk 7]
```diff
@@ -19,32 +19,35 @@ public function __construct(ManagerRegistry $registry)
...
             ->getQuery()
-            ->getResult()
-        ;
+            ->getResult();
     }
-    */
 
-    /*
-    public function findOneBySomeField($value): ?OrderLogsEntity
+    public function getCurrentStageDate(int $orderId): ?array
```
Comment (ralsalloum) [inline backend/src/Repository/OrderLogsEntityRepository.php]: This function seems it returns the last status that the order take, but from its name it seems that it returns the last **status date** that the order take.

File: backend/src/Service/OrderLogs/OrderLogsService.php [hunk 8]
```diff
@@ -42,6 +48,52 @@ public function createOrderLogsRequest(OrderEntity $order, StoreOwnerBranchEntit
        if($branch) {
          $request->setStoreOwnerBranch($branch);
        }
+       
        return $this->createOrderLogs($request);
     }
-}
+
+    public function getOrderLogsByOrderId($orderId): ?array
```
Comment (ralsalloum) [inline backend/src/Service/OrderLogs/OrderLogsService.php:55]: type hint is missing.

File: backend/src/Service/OrderLogs/OrderLogsService.php [hunk 9]
```diff
@@ -42,6 +48,52 @@ public function createOrderLogsRequest(OrderEntity $order, StoreOwnerBranchEntit
...
+        $createDate = "";
+        $orderReceivedDate = "";
+
+        foreach ($orderLogs as $orderLog) {
+            if($orderLog['orderState'] === OrderStateConstant::ORDER_STATE_PENDING) {
+               $createDate = $orderLog['createdAt'];
+            }
+
+            if($orderLog['orderState'] === OrderStateConstant::ORDER_STATE_ON_WAY) {
+               $orderReceivedDate = $orderLog['createdAt'];
```
Comment (ralsalloum) [inline backend/src/Service/OrderLogs/OrderLogsService.php:76]: how you calculate orderReceivedDate while its status still ON WAY?

Comment (habibdevsy) [reply backend/src/Service/OrderLogs/OrderLogsService.php:76]: When the captain receives the order, the order status is on way
Therefore, this case indicates the beginning of receiving the order

**Review by ralsalloum — APPROVED** [review]

**Pull request merged**